src/adapter-manager.js

```diff
@@ -430,24 +432,28 @@ class AdapterManager extends EventEmitter {
         // adapter (i.e. for testing)
         const pluginClient = new PluginClient(manifest.name,
                                               {verbose: false});
-        pluginClient.register().then(adapterManagerProxy => {
+        try {
+          const adapterManagerProxy = await pluginClient.register();
           console.log('Loading adapter', manifest.name, 'as plugin');
-          const adapterLoader = dynamicRequire(path.join(adapterPath));
+          const adapterLoader = dynamicRequire(adapterPath);
           adapterLoader(adapterManagerProxy, newSettings, errorCallback);
-        }).catch(err => {
+        } catch (err) {
           console.error('Failed to register adapter with gateway');
           console.error(err);
-        });
+          return Promise.reject();
+        }
       } else {
         // This is the normal plugin adapter case, and we currently
         // don't need to do anything. We assume that the adapter is
         // started elsewhere.
       }
     } else {
       // Load this adapter directly into the gateway.
-      const adapterLoader = dynamicRequire(path.join(adapterPath));
+      const adapterLoader = dynamicRequire(adapterPath);
       adapterLoader(this, newSettings, errorCallback);
     }
+
+    return Promise.resolve();
```

Comment:
You don't need this return statement, the async function returns a promise which resolves to whatever was returned from the function which is already `undefined`